src/data/resolvers/coinMachine.ts

```diff
@@ -99,5 +99,40 @@ export const coinMachineResolvers = ({
         return null;
       }
     },
+    async currentPeriodTokens(_, { colonyAddress }) {
+      try {
+        const { networkClient } = colonyManager;
+
+        const coinMachineClient = await colonyManager.getClient(
+          ClientType.CoinMachineClient,
+          colonyAddress,
+        );
+
+        const maxPerPeriodTokens = await coinMachineClient.getMaxPerPeriod();
+
+        const activeSoldTokens = await coinMachineClient.getActiveSold();
+        const activePeriod = await coinMachineClient.getActivePeriod();
+        const blockTime = await getBlockTime(networkClient.provider, 'latest');
+
+        const periodLength = await coinMachineClient.getPeriodLength();
+
+        const currentPeriod = Math.floor(
+          bigNumberify(blockTime).div(periodLength.mul(1000)).toNumber(),
+        );
+
+        // eslint-disable-next-line max-len
+        const targetPerPeriodTokens = await coinMachineClient.getTargetPerPeriod();
+
+        return {
+          maxPerPeriodTokens: maxPerPeriodTokens.toString(),
+          activeSoldTokens:
+            activePeriod === currentPeriod ? activeSoldTokens.toString() : '',
```

Comment:
Its not enough, you need to check if there was any TokensBought event in current period, if so then activeSoldTokens if not, then zero